src/backends/glvisualize.jl

```diff
@@ -0,0 +1,90 @@
+
+
+# [WEBSITE]
+
+# ---------------------------------------------------------------------------
+
+immutable GLScreenWrapper
+    window
+    render
+end
+
+function _create_plot(pkg::GLVisualizePackage; kw...)
+  d = Dict(kw)
+  # TODO: create the window/canvas/context that is the plot within the backend (call it `o`)
+  # TODO: initialize the plot... title, xlabel, bgcolor, etc
+  w,r=GLVisualize.glscreen()
+  @async r()
+  Plot(GLScreenWrapper(w,r), pkg, 0, d, Dict[])
+end
+
+
+function _add_series(::GLVisualizePackage, plt::Plot; kw...)
+  d = Dict(kw)
+  # TODO: add one series to the underlying package
+  push!(plt.seriesargs, d)
+  x,y,z=map(Float32,d[:x]), map(Float32,d[:y]), map(Float32,d[:surface].surf)
+  GLVisualize.view(GLVisualize.visualize(x*ones(y)', ones(x)*y', z, :surface))
```

Comment:
The distinction between changing internals and displaying may be best seen with something like plotly, or gadfly. In gadfly, you create the gadfly objects/structure in the "internals" methods, and call draw, etc in the display methods. 

For this case, if there's no underlying internals, then you could consider moving most/all of the logic to the display methods, and just loop through plotargs and seriesargs. 

> On Nov 28, 2015, at 8:42 PM, Simon <notifications@github.com> wrote:
> 
> In src/backends/glvisualize.jl:
> 
> > +function _create_plot(pkg::GLVisualizePackage; kw...)
> > +  d = Dict(kw)
> > +  # TODO: create the window/canvas/context that is the plot within the backend (call it `o`)
> > +  # TODO: initialize the plot... title, xlabel, bgcolor, etc
> > +  w,r=GLVisualize.glscreen()
> > +  @async r()
> > +  Plot(GLScreenWrapper(w,r), pkg, 0, d, Dict[])
> > +end
> > +
> > +
> > +function _add_series(::GLVisualizePackage, plt::Plot; kw...)
> > +  d = Dict(kw)
> > +  # TODO: add one series to the underlying package
> > +  push!(plt.seriesargs, d)
> > +  x,y,z=map(Float32,d[:x]), map(Float32,d[:y]), map(Float32,d[:surface].surf)
> > +  GLVisualize.view(GLVisualize.visualize(x*ones(y)', ones(x)*y', z, :surface))
> Visualize creates a viewable object. Since it already uploads some data to the GPU, sadly a window (OpenGL context) needs to be present already. This is actually addressed in next², where visualize won't convert anything to OpenGL objects (if you don't pass any yourself), so it will be easier to differentiate between setup and display.
> 
> _add_series is for updating the internals for a new data serie
> 
> What does this actually mean? You already plotted e.g. 10 points, and want to update the positions of the points, but leave labels and axis alone?
> 
> —
> Reply to this email directly or view it on GitHub.
>